### PR TITLE
refactor: generate 'canonical_facts' from Parsers instead of Specs

### PR DESCRIPTION
**Pull request by xiangce**
- See RHINENG-8789

### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [x] Is this PR an enhancement?

### CODE REVIEW

File: insights/parsers/ip.py [hunk 1]
```diff
@@ -732,3 +735,37 @@ class IpLinkInfo(IpAddr):
         TX dropped: 0
     """
     pass
+
+
+@parser(Specs.ip_addresses)
+class IPs(CommandParser):
```
Comment (JoySnow) [inline insights/parsers/ip.py:741]: Need to mention this new Parser in this file's docstrings part - Line#18 .

File: insights/parsers/etc_machine_id.py [hunk 2]
```diff
@@ -24,10 +25,13 @@ class EtcMachineId(Parser):
...
+        '4f3fbfda-59ff-4aea-9692-55a73342d439'
     """
 
     def parse_content(self, content):
         content = get_active_lines(content)
         if not content or len(content) != 1:
             raise SkipComponent('Invalid Content')
-        self.machine_id = content[0]
+        self.id = self.machine_id = content[0]
+        self.uuid = str(uuid.UUID(self.id))
```
Comment (JoySnow) [inline insights/parsers/etc_machine_id.py:37]: `uuid.UUID()` could throw error when failed: 
```
raise ValueError('badly formed hexadecimal UUID string')
```

Should we give it an exception handling of this `uuid.UUID()` validation?

Comment (JoySnow) [reply insights/parsers/etc_machine_id.py:37]: Same for `insights/parsers/client_metadata.py` and `insights/parsers/subscription_manager.py` .

**Review by JoySnow — CHANGES_REQUESTED** [review]
Hi @xiangce , the changes LGTM, except the following points:

File: insights/parsers/etc_machine_id.py (hunk 2 above)
Comment (xiangce) [reply insights/parsers/etc_machine_id.py:37]: The "ValueError" will also result in the parser failure, just like the "ContentException", I think it's a bit of "draw legs on a snake" if we catch the "valueError" here and re-raise it with "ContentException.

**Review by JoySnow — APPROVED** [review]
Thanks for the update @xiangce . The changes LGTM.

**Pull request merged**